### ISSUE TITLE
Custom resolver

### ISSUE THREAD

**Issue by vincent-herlemont** [Keats/tera#659]
I have to create a custom resolver. This one has two sources: `embedded` (in the binary) and `local` (the file system).

Example:

Embedded parent template:
```
{% block title %} parent title {% endblock title %}
```  

Local child template: 
```
{% extends "embedded://template" %}
{% block title %} child title {% endblock title %}
``` 
Currently, I use the method [`add_raw_template`]( https://tera.netlify.app/docs/#loading-templates-from-strings) to add different templates, but following this way I have to create my own resolver.

There is another way to use the resolver used by the method [`Tera::new`](https://docs.rs/tera/1.12.1/src/tera/tera.rs.html#88) to support custome resolver?

**Comment by Keats** [comment]
No custom resolvers no.
Do you need to differentiate the origin?

**Comment by vincent-herlemont** [comment]
>  Do you need to differentiate the origin?

Yes. Like that:
- `file://<template>` and local file system.
- `embedded://<template>` and embedded, file embedded in the binary.

**Comment by Keats** [comment]
Yes but why not load all of them in one instance and just extend `"template"`? What I meant by `Do you need to differentiate the origin?` is can you have 2 templates with the same name and do you need to be able to refer to one or the other?
Otherwise there is no plan on having custom resolvers, you can use globwalk and rename manually if you need.

**Comment by Keats** [comment]
Closing as part of the Tera v2 release. If this is still relevant in v2, please reopen — see the 
  migration guide: https://github.com/Keats/tera/blob/master/MIGRATION.md
